### ISSUE TITLE
Error while running Fitnesse Test

### ISSUE THREAD

**Issue by karanbir** [unclebob/fitnesse#332]
Hi,

I have upgraded fitnesse version to 20130531
Afer running any test I am getting following error 

org.apache.velocity.exception.MethodInvocationException: Invocation of method 'execute' in class fitnesse.responders.run.TestResponder$TestExecutor threw exception java.lang.OutOfMemoryError: Java heap space at testPage.vm[line 4, column 15] org.apache.velocity.runtime.parser.node.ASTMethod.handleInvocationException(ASTMethod.java:261) org.apache.velocity.runtime.parser.node.ASTMethod.execute(ASTMethod.java:187) org.apache.velocity.runtime.parser.node.ASTReference.execute(ASTReference.java:280) org.apache.velocity.runtime.parser.node.ASTReference.render(ASTReference.java:369) org.apache.velocity.runtime.parser.node.SimpleNode.render(SimpleNode.java:342) org.apache.velocity.runtime.directive.Parse.render(Parse.java:260) org.apache.velocity.runtime.parser.node.ASTDirective.render(ASTDirective.java:207) org.apache.velocity.runtime.parser.node.SimpleNode.render(SimpleNode.java:342) org.apache.velocity.Template.merge(Template.java:356) org.apache.velocity.Template.merge(Template.java:260) fitnesse.responders.templateUtilities.HtmlPage.render(HtmlPage.java:80) fitnesse.responders.run.TestResponder.doSending(TestResponder.java:63) fitnesse.responders.ChunkingResponder.startSending(ChunkingResponder.java:66) fitnesse.http.ChunkedResponse.sendTo(ChunkedResponse.java:25) fitnesse.FitNesseExpediter.sendResponse(FitNesseExpediter.java:96) fitnesse.FitNesseExpediter.start(FitNesseExpediter.java:48) fitnesse.FitNesseServer.serve(FitNesseServer.java:24) fitnesse.FitNesseServer.serve(FitNesseServer.java:17) fitnesse.socketservice.SocketService$ServerRunner.run(SocketService.java:109) java.lang.Thread.run(Unknown Source) 

Also, at console I am getting following error

 [java] Sep 19, 2013 11:13:27 AM org.apache.velocity.runtime.log.CommonsLogLogChute log
 [java] SEVERE: Exception rendering #parse(testPage.vm) at skeleton.vm[line 45, column 4]

Please suggest solution to resolve the same

Thanks & Regards
Karan

**Comment by RajezMariner** [comment]
@amolenaar Even I have got the same issue. tried running the fitnesse page like this. But I couldn`t get rid of this. Do you have any idea?

**Comment by keetron** [comment]
If you get an out of memory error and you allocate more memory to the JVM, and you still get an out of memory error... Try increasing the memory more! Remember, this is from almost half a decade ago!

`
java -Xmx1024M -jar fitnesse-standalone.jar
`

**Comment by RajezMariner** [comment]
I have tried that.. same issues with 1024M

**Comment by keetron** [comment]
Well, turn it up more!
How much memory would your machine have to spare?

**Comment by RajezMariner** [comment]
2GB is free.

**Comment by keetron** [comment]
well, what happens if you allocate it all to the jvm?
What kind of test did you build anyway that it eats so much space?

**Comment by woodybrood** [comment]
Out of curiosity, does this happen when you run a single test, or is this when you run a suite? Either way, how "big" is the test or the suite?

Years ago, the place where I worked had a huge, and slightly wordy collection of tests. We found that we could not run them all as a set, but had to break them up into chucks. Otherwise we ran slow and could run out of memory due to the large amount of processing required to parse all of those tests and to build the results.
